packages/noya-state/src/reducers/canvasReducer.ts

```diff
@@ -19,8 +29,158 @@ import {
   CompassDirection,
   InteractionAction,
   interactionReducer,
+  InteractionState,
 } from './interactionReducer';
 
+function getAxisValues(
+  rectBounds: Bounds,
+  axis: Axis,
+): [number, number, number] {
+  if (axis === 'x') {
+    // return [rectBounds.minX] as any;
+    return [rectBounds.minX, rectBounds.midX, rectBounds.maxX];
+  } else {
+    // return [] as any;
+    return [rectBounds.minY, rectBounds.midY, rectBounds.maxY];
+  }
+}
+
+function getVisibleLayersAxisValues(
+  selectedLayerIds: string[],
+  state: ApplicationState,
+  interactionState: Extract<InteractionState, { type: 'moving' }>,
+) {
+  const { canvasSize } = interactionState;
+
+  const page = getCurrentPage(state);
+
+  const allVisibleLayers = getLayersInRect(
+    state,
+    {
+      left: 0,
+      right: 0,
+    },
+    createRect({ x: canvasSize.width, y: 0 }, { x: 0, y: canvasSize.height }),
+    {
+      clickThroughGroups: false,
+      includeHiddenLayers: false,
+      includeArtboardLayers: true,
+    },
+  );
+  let layerIsInArtboard = false;
+  let artboardLayers: any[] = [];
+
+  allVisibleLayers
+    .filter((layer) => layer._class === 'artboard' || layer._class === 'group')
+    //TODO : why does TS not recognize artboard.layers ?
```

Comment:
I couldn't figure out why TS couldn't recognize the type of artboard, any suggestions?